### PR TITLE
Overhaul of test_matching.py

### PR DESCRIPTION
**Pull request by Mjh9122**

### CODE REVIEW

File: networkx/algorithms/bipartite/tests/test_matching.py [hunk 1]
```diff
@@ -33,188 +26,95 @@ def setup_method(self):
         self.graph.add_nodes_from(range(12))
         self.graph.add_edges_from(edges)
 
-        # Example bipartite graph from issue 2127
```
Comment (Mjh9122) [inline networkx/algorithms/bipartite/tests/test_matching.py:36]: Changed out complicated specific test case that is disconnected with a separate one that is more simple.

File: networkx/algorithms/bipartite/tests/test_matching.py [hunk 2]
```diff
@@ -33,188 +26,95 @@ def setup_method(self):
...
-        G.add_edge((1, "D"), (0, "A"))
-        G.add_edge((1, "I"), (0, "A"))
-        G.add_edge((1, "I"), (0, "E"))
-        G.add_edge((0, "A"), (1, "G"))
-        G.add_edge((0, "A"), (1, "H"))
-        G.add_edge((0, "E"), (1, "G"))
-        G.add_edge((0, "E"), (1, "H"))
-        self.disconnected_graph = G
-
-    def check_match(self, matching):
```
Comment (Mjh9122) [inline networkx/algorithms/bipartite/tests/test_matching.py:82]: Not needed, more complicated than simply checking the matching in the tests.

File: networkx/algorithms/bipartite/tests/test_matching.py [hunk 3]
```diff
@@ -33,188 +26,95 @@ def setup_method(self):
...
-        matched_vertices = frozenset(itertools.chain(*M.items()))
-        # Assert that the maximum number of vertices (10) is matched.
-        assert matched_vertices == frozenset(range(12)) - {1, 10}
-        # Assert that no vertex appears in two edges, or in other words, that
-        # the matching (u, v) and (v, u) both appear in the matching
-        # dictionary.
-        assert all(u == M[M[u]] for u in range(12) if u in M)
 
-    def check_vertex_cover(self, vertices):
-        """Asserts that the given set of vertices is the vertex cover we
```
Comment (Mjh9122) [inline networkx/algorithms/bipartite/tests/test_matching.py:98]: Also unnecessary, and overly complex, moved to test

File: networkx/algorithms/bipartite/tests/test_matching.py [hunk 4]
```diff
@@ -33,188 +26,95 @@ def setup_method(self):
...
+        assert hopcroft_karp_matching(self.simple_graph) == self.simple_solution
 
     def test_eppstein_matching_disconnected(self):
         with pytest.raises(nx.AmbiguousSolution):
-            match = eppstein_matching(self.disconnected_graph)
+            eppstein_matching(self.disconnected_graph)
 
     def test_hopcroft_karp_matching_disconnected(self):
         with pytest.raises(nx.AmbiguousSolution):
-            match = hopcroft_karp_matching(self.disconnected_graph)
```
Comment (Mjh9122) [inline networkx/algorithms/bipartite/tests/test_matching.py:145]: Redundant, don't hit any new branches/lines

File: networkx/algorithms/bipartite/tests/test_matching.py [hunk 5]
```diff
@@ -33,188 +26,95 @@ def setup_method(self):
...
+            object(),
+        )
         G = nx.Graph([(a, d), (b, d), (b, e), (c, d)])
         matching = maximum_matching(G)
         vertex_cover = to_vertex_cover(G, matching)
         for u, v in G.edges():
             assert u in vertex_cover or v in vertex_cover
 
 
-def test_eppstein_matching():
```
Comment (Mjh9122) [inline networkx/algorithms/bipartite/tests/test_matching.py:207]: Redundant

File: networkx/algorithms/bipartite/tests/test_matching.py [hunk 6]
```diff
@@ -16,14 +16,7 @@
 class TestMatching:
     """Tests for bipartite matching algorithms."""
 
-    def setup_method(self):
```
[the remark this answers is not in the document]
Comment (rossbar) [reply networkx/algorithms/bipartite/tests/test_matching.py:19]: The name of the method should be unchanged (i.e. it must be `setup_method`, not `setup`) as pytest is dropping support for old nose-style test classes in the future.

**Review by rossbar — COMMENTED** [review]
Thanks for the proposal @Mjh9122 . Generally refactoring/removing tests is a delicate procedure as you need very careful review to guarantee that the actual test cases themselves have not change. Generally I'm -1 on the proposed changes here, if for no other reason that it's a lot of churn for seemingly little payoff (with high potential for downside by removing/subtly altering test cases that we very much want to keep).

**Pull request closed without merging**